### PR TITLE
Attempt to clean up use of bad exec()

### PR DESCRIPTION
**Pull request by markxnelson**
This PR attempts to resolve a number of test failures that occur due
to the bad use of the TestUtils.exec() method.  This method throws a
RuntimeException if the return code of the executed command is non-zero.
This method is being used in many places (over 40).  Many of those do
not appear to need or want that behavior, and should instead be using
the ExecCommand.exec() method which executes a command and then returns
the result, without throwing an exception if the command fails.

In this PR, I have done a visual inspection of each use of
TestUtils.exec() and changed them to ExecCommand.exec() where it
appeared to make sense.  This includes every case where the call was
using the return value from the method, and those where it appears clear
from the subsequent logic that a failed command was not expected to
terminate the process.

There were some calls that I left because I could not be sure of the
intent, and these will require more study.

In order to make the code clearer, and make people think twice about
using TestUtils.exec(), which has caused so many failures, I renamed
that method to TestUtils.execOrAbortProcess().  Hopefully that will make
it clearer that that method has a dangerous and unexpected side effect.

Note: This PR does include unmerged changes from PR 1601

### CODE REVIEW

**Comment by markxnelson** [comment]
Queued up a build https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-quicktest/1389/

File: integration-tests/src/test/java/oracle/kubernetes/operator/ItElasticLogging.java [hunk 1]
```diff
@@ -514,12 +516,14 @@ private String execSearchQuery(String queryCriteria, String index)
             .append("'")
             .toString();
     LoggerHelper.getLocal().log(Level.INFO, "Command to search: " + cmd);
-    ExecResult result = TestUtils.exec(cmd);
+    ExecResult result = TestUtils.execOrAbortProcess(cmd);
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/kubernetes/operator/ItElasticLogging.java:519]: we looked into this test yesterday, this should be ExecCommand.exec() ?

**Comment by vanajamukkara** [comment]
LGTM except one I commented above and also some places replaced with ExecCommand.exec() where TestUtils.execOrAbortProcess() can also work, this is fine ASA the method throws Exception.

**Comment by markxnelson** [comment]
test run has 81 passing, 12 failing
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-quicktest/1389/testReport/

**Pull request merged**